demo/site/server.ts

```diff
@@ -53,6 +53,13 @@ app.prepare().then(() => {
                 };
             }
 
+            if (req.headers["rsc"] !== undefined || req.headers["next-router-prefetch"] !== undefined) {
+                if (req.url && new URL(req.url, `http://${req.headers.host}`).searchParams.get("_rsc") === null) {
+                    res.statusCode = 404;
+                    res.end();
```

Comment:
```suggestion
                    return res.end();
```

Don't we need a return here?